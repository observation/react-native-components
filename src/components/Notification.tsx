import React from 'react'
import { View, Text, ViewStyle, StyleProp } from 'react-native'

import font from '../styles/font'
import theme from '../styles/theme'

type Props = {
  count: number
  style?: StyleProp<ViewStyle>
}

const Notification = ({ count, style }: Props) => (
  <View
    style={[
      style,
      {
        backgroundColor: theme.color.error,
        borderRadius: 9,
        width: 18,
        height: 18,
        justifyContent: 'center',
        alignItems: 'center',
      },
    ]}
  >
    <Text allowFontScaling={false} style={{ ...font.extraSmall, color: 'white', textAlign: 'center' }}>
      {count}
    </Text>
  </View>
)

export default Notification
