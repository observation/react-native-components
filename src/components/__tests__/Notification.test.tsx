import React from 'react'

import { render } from '@testing-library/react-native'

import Notification from '../Notification'

describe('Notification', () => {
  test('Rendering', () => {
    // WHEN
    const { toJSON } = render(<Notification count={10} style={{ flex: 1 }} />)

    // THEN
    expect(toJSON()).toMatchSnapshot()
  })
})
