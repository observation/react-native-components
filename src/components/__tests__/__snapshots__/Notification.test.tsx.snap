// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Rendering 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "alignItems": "center",
        "backgroundColor": "#EA554B",
        "borderRadius": 9,
        "height": 18,
        "justifyContent": "center",
        "width": 18,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "white",
        "fontFamily": "Ubuntu",
        "fontSize": 10,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "lineHeight": 16,
        "textAlign": "center",
      }
    }
  >
    10
  </Text>
</View>
`;
